shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate.h"
+
+#include "flutter_accessibility.h"
+
+namespace ui {
+
+//------------------------------------------------------------------------------
+/// Use this class to maintain an accessibility tree. This class consumes
+/// semantics updates from the embedder API and produces an accessibility tree
+/// in the native format.
+///
+/// To use this class, you must provide your own implementation of
+/// FlutterAccessibility and AccessibilityBridgeDelegate.
+class AccessibilityBridge : public AXTreeObserver {
+ public:
+  //------------------------------------------------------------------------------
+  /// Delegate to handle accessibility event and route accessibility action
+  /// back to the Flutter framework.
+  ///
+  /// The accessibility events are generated when accessibility tree changes.
+  /// These events must be sent to the native accessibility system through
+  /// the native API for the system to pick up the changes
+  /// (e.g. NSAccessibilityPostNotification in MacOS).
+  ///
+  /// The accessibility actions are generated by the native accessibility system
+  /// when users interacted with the assistive technologies. Those actions
+  /// needed to be sent to the Flutter framework.
+  class AccessibilityBridgeDelegate {
+   public:
+    virtual ~AccessibilityBridgeDelegate() = default;
+    //------------------------------------------------------------------------------
+    /// @brief      Handle accessibility events generated due to accessibility
+    /// tree
+    ///             changes.
+    ///
+    /// @param[in]  targeted_event      The object that contains both the
+    /// generated
+    ///                                 event and the event target.
+    /// @param[in]  bridge              The pointer to the accessibility bridge
+    /// that
+    ///                                 can be used for querying the
+    ///                                 accessibility information at the time
+    ///                                 the event is fired.
+    virtual void OnAccessibilityEvent(
+        AXEventGenerator::TargetedEvent targeted_event,
+        AccessibilityBridge* bridge) = 0;
+
+    //------------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    /// target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    /// action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(uint16_t target,
+                                             FlutterSemanticsAction action,
+                                             uint8_t* data,
```

Comment:
Please don't use raw pointers like this - I suspect this one wants to be a unique_ptr